pkg/apis/core/v1alpha1/federationconfig_types.go

```diff
@@ -26,56 +26,68 @@ type FederationConfigSpec struct {
 	// The scope of the federation control plane should be either `Namespaced` or `Cluster`.
 	// `Namespaced` indicates that the federation namespace will be the only target for federation.
 	Scope              apiextv1b1.ResourceScope `json:"scope,omitempty"`
-	ControllerDuration DurationConfig           `json:"controller-duration,omitempty"`
-	LeaderElect        LeaderElectConfig        `json:"leader-elect,omitempty"`
-	FeatureGates       []FeatureGatesConfig     `json:"feature-gates,omitempty"`
-	ClusterHealthCheck ClusterHealthCheckConfig `json:"cluster-health-check,omitempty"`
-	SyncController     SyncControllerConfig     `json:"sync-controller,omitempty"`
+	ControllerDuration DurationConfig           `json:"controllerDuration,omitempty"`
```

Comment:
I'm guessing this is WIP for more than just the test failure, and you're going to make these pointers and add `+optional`?

Reply:
Correct, this is WIP because there's still more work to do.